smartsim/settings/pbsSettings.py

```diff
@@ -168,7 +194,12 @@ def set_resource(self, resource_name: str, value: str) -> None:
         """
         # TODO add error checking here
         # TODO include option to overwrite place (warning for orchestrator?)
-        self.resources[resource_name] = value
+        self._resources[resource_name] = value
+        self._sanity_check_resources()
+        # Capture the case where someone is setting the number of nodes
+        # through 'select' or 'nodes'
+        if resource_name in ["select", "nodes"] and value:
+            self._nodes = int(value)
```

Comment:
It seems like `_nodes` is not accessed anywhere else, is this assignment still needed?